test/e2e/activator_test.go

```diff
@@ -124,3 +128,105 @@ func TestActivatorOverload(t *testing.T) {
 	}
 	group.Wait()
 }
+
+func TestActivatorRevisionTimeout(t *testing.T) {
+	t.Parallel()
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	const (
+		// The number of concurrent requests to hit the activator with.
+		concurrency = 1
+		// How long the service will process the request in ms.
+		serviceSleep = 10000
+		// body of timeout response from activator
+		wantTimeoutRespBody = "activator request timeout"
+	)
+	var (
+		group       sync.WaitGroup
+		resp        []*spoof.Response
+		mux         sync.Mutex
+		timeoutResp = 0
+		successResp = 0
+	)
+	clients := Setup(t)
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "timeout",
+	}
+
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer test.TearDown(clients, names)
+
+	t.Log("Creating a service with run latest configuration.")
+	// Create a service with concurrency 1 that sleeps for N ms.
+	// Limit its maxScale to 10 containers, wait for the service to scale down and hit it with concurrent requests.
+	resources, err := v1test.CreateServiceReady(t, clients, &names,
+		rtesting.WithContainerConcurrency(1),
+		rtesting.WithRevisionTimeoutSeconds(15),
+		rtesting.WithConfigAnnotations(map[string]string{
+			autoscaling.TargetBurstCapacityKey: "-1",
+			autoscaling.MaxScaleAnnotationKey:  "1",
+		}))
+	if err != nil {
+		t.Fatal("Unable to create resources:", err)
+	}
+
+	deploymentName := rnames.Deployment(resources.Revision)
+	if err := WaitForScaleToZero(t, deploymentName, clients); err != nil {
+		t.Fatalf("Unable to observe the Deployment named %s scaling down: %v", deploymentName, err)
+	}
+
+	domain := resources.Route.Status.URL.Host
+	client, err := pkgTest.NewSpoofingClient(clients.KubeClient, t.Logf, domain, test.ServingFlags.ResolvableDomain, test.AddRootCAtoTransport(t.Logf, clients, test.ServingFlags.Https))
+	if err != nil {
+		t.Fatal("Error creating the Spoofing client:", err)
+	}
+
+	url := fmt.Sprintf("http://%s/?timeout=%d", domain, serviceSleep)
+
+	t.Log("Starting to send out requests")
+
+	// Send requests async and wait for the responses.
+	for i := 0; i < 5; i++ {
+		group.Add(1)
+		go func(i int) {
+			defer group.Done()
+			// We need to create a new request per HTTP request because
+			// the spoofing client mutates them.
+			req, err := http.NewRequest(http.MethodGet, url, nil)
+			if err != nil {
+				t.Errorf("error creating http request: %v", err)
+			}
+			res, err := client.Do(req)
+			if err != nil {
+				t.Errorf("unexpected error sending a request, %v", err)
+				return
+			}
+			mux.Lock()
+			resp = append(resp, res)
```

Comment:
nit: if you pre-allocate the `resp` with capacity 5 (i.e. `resp := make([]*spoof.Response, 5)`), then you can use `i` as index (i.e. `resp[i] = res`), and you don't have to use Lock here.

Reply:
Thanks, done